bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/Workbench.java

```diff
@@ -306,16 +308,18 @@ private static final class StartupProgressBundleListener implements SynchronousB
 			displayForStartupListener = display;
 			subMonitor = SubMonitor.convert(progressMonitor);
 			subMonitor.setTaskName(NLS.bind(WorkbenchMessages.Startup_Loading, Platform.getProduct().getName()));
+			BundleContext context = WorkbenchPlugin.getDefault().getBundleContext();
+			context.addServiceListener(new ServiceListener() {
+				@Override
+				public void serviceChanged(ServiceEvent event) {
+					subMonitor.setWorkRemaining(5).worked(1);
+					spinEventQueueToUpdateSplash(displayForStartupListener);
+				}
+			});
 		}
 
 		@Override
 		public void bundleChanged(BundleEvent event) {
```

Comment:
If this method becomes empty - delete it and let the class not implements SynchronousBundleListener

Reply:
Thanks @akurtakov I have addressed this change which resulted in a number of useful other changes. Now it looks clean!.